pkg/udf/forward/forward.go

```diff
@@ -297,13 +297,15 @@ func (isdf *InterStepDataForward) forwardAChunk(ctx context.Context) error {
 		// Trigger the UDF processing based on the mode enabled for map
 		// ie Batch Map or unary map
 		// This will be a blocking call until the all the UDF results for the batch are received.
+		udfStart := time.Now()
 		udfResults, err = isdf.applyUDF(ctx, dataMessages)
 		if err != nil {
 			isdf.opts.logger.Errorw("failed to applyUDF", zap.Error(err))
 			// As there's no partial failure, non-ack all the readOffsets
 			isdf.fromBufferPartition.NoAck(ctx, readOffsets)
 			return err
 		}
+		metrics.UDFProcessingTime.With(metricLabels).Observe(float64(time.Since(udfStart).Microseconds()))
```

Comment:
should this be `ConcurrentUDFProcessingTime`?

Reply:
It would be in batch mode, but how about in unary mode? In my understanding, calling applyUDF() will blocks until all the UDF results are received.

Is it better that we visit either concurrent or non-concurrent metric, depending on the mode?